.clang-tidy

```diff
@@ -26,9 +26,18 @@ Checks:     '
             performance-*,
             portability-*,
             readability-*,
+            -bugprone-exception-escape,
+            -bugprone-macro-parentheses,
+            -bugprone-too-small-loop-variable,
+            -clang-analyzer-cplusplus.NewDelete,
+            -clang-analyzer-cplusplus.NewDeleteLeaks,
             -google-readability-braces-around-statements,
-            -google-readability-namespace-comments,
+            -modernize-use-nodiscard,
```

Comment:
Don't fully understand this one, but seems to blow up all over the codebase. We can have a discussion on whether it's worth enforcing this and someone going through and copy-pasting [[nodiscard]] but seems like a waste of time.

Reply:
Here's the relevant information about it:
https://en.cppreference.com/w/cpp/language/attributes/nodiscard

Reply:
My understanding is that it is "warn on unused result". I do think it is useful, but maybe not worth cluttering every function definition with...